include/nng/supplemental/util/nng_snprintf.h

```diff
@@ -0,0 +1,42 @@
+#ifndef NNG_SNPRINTF_H
+#define NNG_SNPRINTF_H
+
+#include <stdarg.h>  // NOLINT
+#include <stdio.h>   // NOLINT
+
+#if !defined(__cplusplus) && defined(_MSC_VER)
+#define inline __inline
+#endif  //__cplusplus && _MSC_VER
+
+#ifdef __cplusplus
+extern "C" {
+#endif  //__cplusplus
+
+#if !defined(_TRUNCATE)
+#define _TRUNCATE ((size_t)-1)
+#endif  //_TRUNCATE
+
+inline int nng_snprintf(char* const _Buffer, int const _BufferCount, char const* const _Format, ...) {
```

Comment:
This by the way doesn't conform to our other coding style (see clang-format).  Having said that, we also don't make use if inline anywhere else in nng today.  I've been cautious about doing thus far -- no where this function is used is sufficiently performance sensitive to justify the effort the stack frame any way.

Reply:
The fact that you have to redefine it as __inline for MSVC is a telling statement btw.  I don't consider inline portable.

Reply:
If you like, I can happily add a source file instead of inline.